amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -296,26 +385,27 @@ private void recordsReceived(RecordFlow triggeringFlow, SubscribeToShardEvent re
             FanoutRecordsRetrieved recordsRetrieved = new FanoutRecordsRetrieved(input,
```

Comment:
How are we getting a hold of retrieved records in onError? This diff doesn't show the diff against original file that is in mainline: https://github.com/awslabs/amazon-kinesis-client/blob/6c64055d9b81b51480ab844be1769a3637c2c29c/amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java#L144

if (!isActiveFlow(triggeringFlow))  condition seems to be new, and original version only has else() condition. It'll be best for this PR to show all changes in a single commit. Maybe I'm missing a feature of github to see the full diff against mainline : )